### PR TITLE
python3Packages.freebox-api: 0.0.9 -> 0.0.10

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 0.0.10

Change log: https://github.com/hacf-fr/freebox-api/releases/tag/v0.0.10

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 118182` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.freebox-api</li>
    <li>python39Packages.freebox-api</li>
  </ul>
</details>

File: pkgs/servers/home-assistant/component-packages.nix [hunk 1]
```diff
@@ -277,7 +277,7 @@
     "foscam" = ps: with ps; [ ]; # missing inputs: libpyfoscam
     "foursquare" = ps: with ps; [ aiohttp-cors ];
     "free_mobile" = ps: with ps; [ ]; # missing inputs: freesms
-    "freebox" = ps: with ps; [ aiohttp-cors netdisco zeroconf ]; # missing inputs: freebox-api
+    "freebox" = ps: with ps; [ aiohttp-cors freebox-api netdisco zeroconf ];
```
Comment (SuperSandro2000) [inline pkgs/servers/home-assistant/component-packages.nix:280]: Can we activate some new tests?

File: pkgs/development/python-modules/freebox-api/default.nix [hunk 2]
```diff
@@ -2,29 +2,38 @@
...
     repo = pname;
     rev = "v${version}";
-    sha256 = "0qn7jqfbp850aqgfsxjfv14myi6idz6sf7024p6wpqpa2xk0vfiq";
+    sha256 = "sha256-yUcHdSHSgWxZl0z7Ue0MestvGhiXkDsxArNoDk0ZkR4=";
   };
 
-  nativeBuildInputs = [ poetry ];
+  nativeBuildInputs = [
+    # Switch to peotry-core, https://github.com/hacf-fr/freebox-api/pull/187
+    poetry
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/freebox-api/default.nix:26]: Can we fetch that PR?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/servers/home-assistant/component-packages.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/servers/home-assistant/component-packages.nix:280]: Yes, we can.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118182` at d1fe52d4 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.freebox-api</li>
    <li>python39Packages.freebox-api</li>
  </ul>
</details>

File: pkgs/development/python-modules/freebox-api/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/development/python-modules/freebox-api/default.nix]: Sure.

File: pkgs/servers/home-assistant/component-packages.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/servers/home-assistant/component-packages.nix:280]: Should be possible.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 118182` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>home-assistant</li>
    <li>python38Packages.freebox-api</li>
    <li>python39Packages.freebox-api</li>
  </ul>
</details>

File: pkgs/development/python-modules/freebox-api/default.nix [hunk 3]
```diff
@@ -2,29 +2,46 @@
...
     rev = "v${version}";
-    sha256 = "0qn7jqfbp850aqgfsxjfv14myi6idz6sf7024p6wpqpa2xk0vfiq";
+    sha256 = "sha256-yUcHdSHSgWxZl0z7Ue0MestvGhiXkDsxArNoDk0ZkR4=";
   };
 
-  nativeBuildInputs = [ poetry ];
+  patches = [
+    (fetchpatch {
+      name = "switch-to-poetry-core.patch";
+      url = "https://github.com/hacf-fr/freebox-api/pull/187/commits/07356ac65483bc24fb1ed32612e77f2c2eed0134.patch";
```
Comment (dotlambda) [inline pkgs/development/python-modules/freebox-api/default.nix:28]:
```suggestion
      url = "https://github.com/hacf-fr/freebox-api/commit/07356ac65483bc24fb1ed32612e77f2c2eed0134.patch";
```

File: pkgs/development/python-modules/freebox-api/default.nix [hunk 4]
```diff
@@ -2,29 +2,46 @@
...
     owner = "hacf-fr";
     repo = pname;
     rev = "v${version}";
-    sha256 = "0qn7jqfbp850aqgfsxjfv14myi6idz6sf7024p6wpqpa2xk0vfiq";
+    sha256 = "sha256-yUcHdSHSgWxZl0z7Ue0MestvGhiXkDsxArNoDk0ZkR4=";
   };
 
-  nativeBuildInputs = [ poetry ];
+  patches = [
+    (fetchpatch {
```
Comment (dotlambda) [inline pkgs/development/python-modules/freebox-api/default.nix:26]:
```suggestion
    # https://github.com/hacf-fr/freebox-api/pull/187
    (fetchpatch {
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
What's the downside of using `poetry` until the upstream pull request is merged?

File: pkgs/development/python-modules/freebox-api/default.nix (hunk 3 above)
Comment (fabaff) [reply pkgs/development/python-modules/freebox-api/default.nix:28]: Done

File: pkgs/development/python-modules/freebox-api/default.nix (hunk 4 above)
Comment (fabaff) [reply pkgs/development/python-modules/freebox-api/default.nix:26]: Added, thanks.

**Comment by mweinelt** [comment]
> What's the downside of using `poetry` until the upstream pull request is merged?

Both approaches are probably fine. Sure, there is a chance that upstream rejects the patch, but I feel that it is slim to none in this case.

**Comment by mweinelt** [comment]
Test failure on x86_64 is unrelated (prometheus humidity sensor test).

**Pull request merged**